### PR TITLE
[REF] Sort participants file by 'participant_id', prevent creation of garbage columns

### LINKED ISSUE #23: Sort 'participants.tsv' by 'participant_id' and clean outputs if multiple sessions appear
Currently the `participants.tsv` file outputted isn't sorted by 'participant_id` and it ends up looking nasty. The second issue with this involves `participant_id` that appear multiple times in a dataset. They end up getting second age, sex, and weight columns. 

image

### PR DESCRIPTION
**Pull request by akimbler**
Closes #23

Changes Proposed:
* Sort `participants.tsv` by `participant_id`.
* Checks for existence of `age, sex, and weight` columns, and doesn't create more if they already exist
* Changed `tmp` to `.tmp` for `tmpdir`

### CODE REVIEW

File: bidsify/bidsify.py [hunk 1]
```diff
@@ -129,11 +129,14 @@ def bidsify_workflow(dicom_dir, heuristics, subject, session=None, output_dir='.
...
         else:
             age = np.nan
-
-        new_participant = pd.DataFrame(columns=['age', 'sex', 'weight'],
-                                       data=[[age, data.PatientSex,
-                                              data.PatientWeight]])
-        participant_df = pd.concat([participant_df, new_participant], axis=1)
+        column_set = set(participant_df.columns)
+        # If participant_df already has age, sex, and weight, skip
+        if not column_set.issuperset({'age', 'sex', 'weight'}):
```
Comment (tsalo) [inline bidsify/bidsify.py]: what if there's partial overlap?

Comment (tsalo) [reply bidsify/bidsify.py]: also should the new participant _not_ be added if the larger df has all three columns?

File: bidsify/bidsify.py [hunk 2]
```diff
@@ -131,18 +132,23 @@ def bidsify_workflow(dicom_dir, heuristics, subject, session=None, output_dir='.
...
+        for mc in missing_cols:
+            participant_df[mc] = np.nan
+        if participant_id not in participant_df.index.values:
+            participant_df.loc[participant_id] = np.nan
+
+        participant_df.update(additional_data, overwrite=True)
+        participant_df.sort_index(inplace=True)
+        participant_df.to_csv(
+            participants_file, sep='\t', na_rep='n/a',
+            line_terminator='\n', index=False, index_label='participant_id')
```
Comment (tsalo) [inline bidsify/bidsify.py]: I think you just need to drop `index=False`.

Comment (akimbler) [reply bidsify/bidsify.py]: Done

**Review by tsalo — APPROVED** [review]
LGTM.

**Pull request merged**